### DISCUSSION TITLE
What do you do when clearing the store?

### DISCUSSION THREAD

**Discussion by 0oooooooo0** [pmndrs/zustand#2183] [Q&A] [answered]
```
import { create } from 'zustand'

interface State {
  text: string
}

const initialState: State = {
  text: '',
}

interface Action {
  setText: (text: string) => void
  reset: () => void
}

export const useTempStore = create<State & Action>(set => ({
  ...initialState,
  setText: text => set({ text }),
  reset: () => set(initialState),
}))

export const tempState = useTempStore.getState
```

I am creating a reset method for each store to clear like this code, but I was wondering if there is another good way to do it.
Also, I was wondering if there is a method to help with the clear?

I also seem to use a lot of repetitive code to clear the store when unmounted. Is it just me?

```
useEffect(() => tempState.reset())
```

**Comment by dbritto-dev** [comment]
> ```
> import { create } from 'zustand'
> 
> interface State {
>   text: string
> }
> 
> const initialState: State = {
>   text: '',
> }
> 
> interface Action {
>   setText: (text: string) => void
>   reset: () => void
> }
> 
> export const useTempStore = create<State & Action>(set => ({
>   ...initialState,
>   setText: text => set({ text }),
>   reset: () => set(initialState),
> }))
> 
> export const tempState = useTempStore.getState
> ```
> 
> I am creating a reset method for each store to clear like this code, but I was wondering if there is another good way to do it. Also, I was wondering if there is a method to help with the clear?
> 
> I also seem to use a lot of repetitive code to clear the store when unmounted. Is it just me?
> 
> ```
> useEffect(() => tempState.reset())
> ```

@0oooooooo0 that's OK, just one little change

```ts
reset: () => set(initialState, true)
```

**Reply by 0oooooooo0** [reply]
Why is the second argument true?

**Reply by dbritto-dev** [reply]
@0oooooooo0 to replace the whole state instead of shallow merge it

**Reply by 0oooooooo0** [reply]
Doesn't that make setText disappear?

**Reply by dbritto-dev** [reply, marked as answer]
@0oooooooo0 yeah, you are right; my bad. Here you can find more about that -> https://docs.pmnd.rs/zustand/guides/how-to-reset-state (go to the codesandbox demo section)

**Reply by dbritto-dev** [reply]
when you reset the state from the store itself you shouldn't use the second parameter but when you reset the state wrapping the create fn in order to reset more than one store then you can use the second parameter due to the initial state is the whole state including the fns

**Reply by dbritto-dev** [reply]
in other to clear the store when is unmounted you should do this:

```ts
const resetTempStore = useTempStore(state => state.reset)

useEffect(() => {
  return () => void resetTempStore()
}, [])
```
